fs/span-manager/span_manager.go

```diff
@@ -38,6 +38,29 @@ var (
 	ErrExceedMaxSpan       = errors.New("span id larger than max span id")
 )
 
+type MultiReaderCloser struct {
+	c []io.Closer
+	io.Reader
+}
+
+func (mrc *MultiReaderCloser) Close() error {
+	for _, c := range mrc.c {
+		if err := c.Close(); err != nil {
+			return err
```

Comment:
I think we want to be sure that we at least try to close all the inner Closers before returning to minimize a leak in the case of error. Can we switch this to a multi-error that we append to and only return after processing all the closers?

Reply:
Agh,  good catch. Will do.

Reply:
I've updated to use `errors.Join`, introduced in Go 1.20 - https://pkg.go.dev/errors#Join. I could of have implemented a separate wrapper Error type or just used `strings.Join` and then returned a new error, but since we already use multi error wrapping introduced in 1.20 (#788),  I thought this made sense.

Reply:
We're using [go-multierror](https://github.com/hashicorp/go-multierror) elsewhere ([e.g.](https://github.com/awslabs/soci-snapshotter/blob/8c6880c62279317ca0e66629f70f70ce99babcc5/fs/reader/reader.go#L191)). I do like the idea of being able to get rid of that as a direct dependency, though. I'll create an issue to replace those with this style.

Reply:
https://github.com/awslabs/soci-snapshotter/issues/793